persistent-postgresql/Database/Persist/Postgresql.hs

```diff
@@ -622,29 +624,39 @@ mkForeignAlt name fdef =
         map escape parentfields
 
 addTable :: [Column] -> EntityDef -> AlterDB
-addTable cols entity = AddTable $ T.concat
-                       -- Lower case e: see Database.Persist.Sql.Migration
-                       [ "CREATe TABLE " -- DO NOT FIX THE CAPITALIZATION!
-                       , escape name
-                       , "("
-                       , idtxt
-                       , if null cols then "" else ","
-                       , T.intercalate "," $ map showColumn cols
-                       , ")"
-                       ]
-    where
-      name = entityDB entity
-      idtxt = case entityPrimary entity of
-                Just pdef -> T.concat [" PRIMARY KEY (", T.intercalate "," $ map (escape . fieldDB) $ compositeFields pdef, ")"]
-                Nothing   ->
-                    let defText = defaultAttribute $ fieldAttrs $ entityId entity
-                        sType = fieldSqlType $ entityId entity
-                    in  T.concat
-                            [ escape $ fieldDB (entityId entity)
-                            , maySerial sType defText
-                            , " PRIMARY KEY UNIQUE"
-                            , mayDefault defText
-                            ]
+addTable cols entity =
+    AddTable $ T.concat
+        -- Lower case e: see Database.Persist.Sql.Migration
+        [ "CREATe TABLE " -- DO NOT FIX THE CAPITALIZATION!
+        , escape name
+        , "("
+        , idtxt
+        , if null nonIdCols then "" else ","
+        , T.intercalate "," $ map showColumn nonIdCols
+        , ")"
+        ]
+  where
+    nonIdCols =
+        filter (\c -> cName c /= fieldDB (entityId entity) ) cols
```

Comment:
We filter out the id column here,because it is treated specially in `idtxt`. And unfortunately it would be pretty difficult/annoying to set a default at the `Column` level because that bit isn't database agnostic.